plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -204,15 +205,20 @@ being "internal".
 
 instance KnownTypeAst DefaultUni Int where
     toTypeAst _ = toTypeAst $ Proxy @Integer
+    {-# INLINE toTypeAst #-}
 
 -- See Note [Int as Integer].
+-- See Note [Performance of KnownTypeIn instances]
 instance HasConstantIn DefaultUni term => KnownTypeIn DefaultUni term Int where
     makeKnown emit mayCause = makeKnown emit mayCause . toInteger
-    readKnown mayCause term = do
-        i :: Integer <- readKnown mayCause term
-        unless (fromIntegral (minBound :: Int) <= i && i <= fromIntegral (maxBound :: Int)) $
-            throwingWithCause _EvaluationFailure () mayCause
-        pure $ fromIntegral i
+    {-# INLINE makeKnown #-}
+
+    readKnown mayCause term =
+        readKnownConstant mayCause term >>= oneShot \(i :: Integer) ->
```

Comment:
I'd put another note reference just above the use of `oneShot` here, since it's particularly notable.

Reply:
Will do.